### DISCUSSION TITLE
Menu positioning breaks with custom button element

### DISCUSSION THREAD

**Discussion by liegeandlief** [ariakit/ariakit#680] [Questions] [answered]
### Current behavior

When I use a custom button element for `MenuButton` the automatic positioning of the menu ceases to work. Looking at the menu's `div` element the `style` attribute toggles between `display: none` when the menu is closed and empty when the menu is open.

The custom button element I am using just renders an HTML button and it also maps all passed in props on to the button element.

### Steps to reproduce the bug

Menu component:

```
import {
  useMenuState,
  Menu,
  MenuItem,
  MenuButton,
  MenuSeparator,
} from "reakit/Menu";

const Button = props => <button {...props} className='exampleCustomClass'>Open menu</button>

function Example() {
  const menu = useMenuState();

  return (
    <>
      <MenuButton {...menu} as={Button}>Menu</MenuButton>
      <Menu {...menu} aria-label="Example">
        <MenuItem
          {...menu}
          onClick={() => {
            menu.hide();
            console.log("clicked on button");
          }}
        >
          Button
        </MenuItem>
        <MenuItem {...menu} as="a" href="#" onClick={menu.hide}>
          Link
        </MenuItem>
      </Menu>
    </>
  );
}
```

In the abov example the menu won't be positioned. Removing the `as` prop from `MenuButton` makes positioning work again.

### Expected behavior

The menu positioning should continue to work as it does when a custom button element is not provided via the `as` prop.

[a comment by an automated account is not shown]

**Comment by diegohaz** [comment]
Hi @liegeandlief! You have to `forwardRef` on your custom Button component. You should've seen a console warning about that.

**Comment by liegeandlief** [comment]
Thanks @diegohaz! Are you able to provide an example of what you mean? Thanks :)

**Comment by diegohaz** [comment, marked as answer]
Sure:
```jsx
const Button = React.forwardRef((props, ref) => (
  <button {...props} ref={ref} className='exampleCustomClass'>Open menu</button>
));
```

**Comment by liegeandlief** [comment]
That worked a treat. Thanks!
